Trying to improve cleanup - (Task #271)

PhilMFischer opened the pull request:
- Calling the clear editing domain registry from test setup and tear
down now.
- Changed registry clear to also shutdown NotificationThread.
- Changed code to wait for thread to be actually shutdown.

---
Task #271: Improve Cleanup of AProject test Case

Commit:
Trying to improve cleanup - (Task #271)

- Calling the clear editing domain registry from test setup and tear
down now.
- Changed registry clear to also shutdown NotificationThread.
- Changed code to wait for thread to be actually shutdown.

---
Task #271: Improve Cleanup of AProject test Case

SaMuellerDLR commented on de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatTransactionalEditingDomain.java:
```diff
@@ -657,7 +657,16 @@ public void run() {
+			triggerFinished = true;
+			
+			// Now wait until the thread actually terminated
+			while (this.isAlive()) {
```
This looks like a re-implementation of the join() method of the Thread class.

PhilMFischer replied:
Right. Done!

SaMuellerDLR commented on de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/test/AProjectTestCase.java:
```diff
@@ -96,8 +98,7 @@ public void tearDown() throws CoreException {
 		}
 
 		if (editingDomain != null) {
```
Do we still need the null check? I think we can simplify this by always calling clear and setting the editingDomain to null regardless.

PhilMFischer replied:
Agree! Removed

Commit:
Merge branch 'development' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_271-Improve_Cleanup_of_AP

Commit:
Refactoring thread tear down - (Task #271)

- using join as suggested.

---
Task #271: Improve Cleanup of AProject test Case

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_271-Improve_Cleanup_of_AP

Commit:
Improved tearDown of AProjectTestCase - (Task #271)

- Followed suggestion of review.

---
Task #271: Improve Cleanup of AProject test Case
